apps/crn-server/test/data-providers/contentful/teams.data-provider.test.ts

```diff
@@ -0,0 +1,552 @@
+            total: 3,
+            items: [
+              {
+                ...getContenfulGraphqlTeam({}),
+                toolsCollection: {
+                  items: brokenUrlTools,
+                },
+              },
+              {
+                ...getContenfulGraphqlTeam({}),
+                toolsCollection: {
+                  items: brokenNameTools,
+                },
+              },
+              getContenfulGraphqlTeam({ tools: fullTools }),
+            ],
+          },
+        });
+
+        const result = await teamDataProvider.fetch({
+          take: 8,
+          skip: 0,
+        });
+        expect(result.items[0]!.tools).toEqual(tools);
+        expect(result.items[1]!.tools).toEqual(tools);
+        expect(result.items[2]!.tools).toEqual(fullTools);
+      });
+    });
+  });
+
+  describe('Fetch-by-id method', () => {
+    test('Should fetch the team from Contentful GraphQl', async () => {
+      const teamId = 'team-id-0';
+      const result = await teamDataProviderMock.fetchById(teamId);
+
+      expect(result).toMatchObject(getContentfulTeamDataObject());
+    });
+
+    test('Should return null when the team is not found', async () => {
+      const teamId = 'not-found';
+
+      const contentfulGraphQLResponse = getContentfulTeamsGraphqlResponse();
+      contentfulGraphQLResponse.teamsCollection = null;
+
+      contentfulGraphqlClientMock.request.mockResolvedValueOnce(
+        contentfulGraphQLResponse,
+      );
+
+      expect(await teamDataProvider.fetchById(teamId)).toBeNull();
+    });
+
+    test('Should throw an error with a specific error message when the graphql client throws one', async () => {
+      const id = 'some-id';
+      contentfulGraphqlClientMock.request.mockRejectedValueOnce(
+        new GraphQLError('some error message'),
+      );
+
+      await expect(teamDataProvider.fetchById(id)).rejects.toThrow(
+        'some error message',
+      );
+    });
+
+    test('Should return the result when the team exists', async () => {
+      const id = 'some-id';
+      const contentfulGraphQLResponse = {
+        teams: getContenfulGraphqlTeam({}),
+      };
+
+      contentfulGraphqlClientMock.request.mockResolvedValueOnce(
+        contentfulGraphQLResponse,
+      );
+
+      const result = await teamDataProvider.fetchById(id);
+
+      expect(result).toEqual(getContentfulTeamDataObject());
+      expect(contentfulGraphqlClientMock.request).toBeCalledWith(
+        expect.anything(),
+        expect.objectContaining({
+          id,
+        }),
+      );
+    });
+  });
+
+  describe('Update method', () => {
+    test('Should update the team', async () => {
+      const teamId = 'team-id-1';
+      const tool = {
+        name: 'Youtube Channel',
+        description: 'Youtube channel with team videos',
+        url: 'http://www.youtube.com/abcde',
+      };
+      const toolMock = getEntry({});
+
+      jest
+        .spyOn(await contentfulRestClientMock, 'createEntry')
```

Comment:
why are you having to spy on a mock that is already injected into the data provider? I see you enjoy jest's mocking functions a lot :D but you could just assign the result of `getContentfulEnvironmentMock()` into a const and make assertions on that